pkg/kubelet/kubelet.go

```diff
@@ -893,8 +894,17 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	return m, nil
 }
 
-// Make the service environment variables for a pod in the given namespace.
-func (kl *Kubelet) makeEnvironmentVariables(ns string, container *api.Container) ([]string, error) {
+func (kl *Kubelet) getDownwardAPIEnvVarMap(pod *api.Pod) map[string]string {
+	return map[string]string{
+		"K8S_API_POD_NAME": pod.Name,
```

Comment:
Sorry for being late to the party. I was out yesterday, and email is out of control, in general. 

As I commented on #2316, I'd like to seriously consider an alternative approach: use the API field names directly, no fake env vars on the RHS.